nautobot/extras/views.py

```diff
@@ -1311,9 +1319,12 @@ def get(self, request, class_path=None, pk=None):
                     # for example "?kwargs_from_job_result=<UUID>&integervar=22"
                     explicit_initial = initial
                     initial = job_result.task_kwargs.copy()
-                    task_queue = job_result.celery_kwargs.get("queue", None)
-                    if task_queue is not None:
-                        initial["_task_queue"] = task_queue
+                    job_queue = job_result.celery_kwargs.get("queue", None)
+                    if job_queue is not None:
+                        jq, _ = JobQueue.objects.get_or_create(
+                            name=job_queue, defaults={"queue_type": JobQueueTypeChoices.TYPE_CELERY}
+                        )
+                        initial["_job_queue"] = jq
```

Comment:
Here too it feels wrong for a `get` to have a side effect of creating a database record.

Reply:
Should we raise an error here JobQueue.DoesNotExist?

Reply:
I would also push back a little on that it emulates the previous behavior to the best of our abilities. Because task_queue is only a simple string and not a database record. Requiring the user to manually create a job queue before passing it to celery_kwargs might break some things.